renderer/viewer/lib/chunkCacheIntegration.ts

```
@@ -0,0 +1,124 @@
+  return (hash >>> 0).toString(16).padStart(8, '0')
+}
+
+/**
+ * Generate a simple hash from block state IDs (async version using crypto.subtle)
+ * Use this for more secure hashing when persistent storage is used
+ */
+export async function computeBlockHashAsync (blockStateIds: Uint16Array): Promise<string> {
+  if (typeof crypto !== 'undefined' && crypto.subtle) {
+    try {
+      const buffer = await crypto.subtle.digest('SHA-256', blockStateIds.buffer)
+      const hashArray = Array.from(new Uint8Array(buffer))
```

ESLint

[error] Prefer the spread operator over `Array.from(…)`.

(unicorn/prefer-spread)